src/algorithms/pathfinders/dijkstra.ts

```diff
@@ -0,0 +1,111 @@
+import EdgeCost from '../models/edgecost';
+import Graph from '../models/graph';
+import Vertex from '../models/vertex';
+/**
+ * Implements Dijkstra's algorithm to find the shortest path between 2 vertices
+ */
+export default class DijkstraPathfinder {
+    /**
+     * Stores best known distances from starting vertex to all the others
+     */
+    distances: Map<number, {
+        distance: number;
+        sourceID: number;
+    }>;
+
+    /**
+     * stores all unvisited vertices
+     */
+    unvisited: Set<number>;

```

Comment:
This is not a good object for storing unvisited vertices, you must use priority queue. Otherwise you need to go through the whole set to find the next vertex and instead of complexity O(1) it's O(n), probably even worse. So just in our small campus example with tens of thousands of vertices it's 10000+x slower than it could be. There is no native priority queue object in JS but you can find many implementations online.